android/src/com/mapswithme/maps/SplashActivity.java

```diff
@@ -0,0 +1,239 @@
+package com.mapswithme.maps;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v4.app.ActivityCompat;
+import android.support.v7.app.AppCompatActivity;
+import android.view.View;
+
+import com.mapswithme.maps.ads.LikesManager;
+import com.mapswithme.maps.editor.ViralFragment;
+import com.mapswithme.maps.location.LocationHelper;
+import com.mapswithme.maps.news.BaseNewsFragment;
+import com.mapswithme.maps.news.FirstStartFragment;
+import com.mapswithme.maps.news.NewsFragment;
+import com.mapswithme.util.Counters;
+import com.mapswithme.util.UiUtils;
+import com.mapswithme.util.Utils;
+import com.mapswithme.util.concurrency.UiThread;
+import com.mapswithme.util.statistics.PushwooshHelper;
+
+import static android.Manifest.permission.ACCESS_COARSE_LOCATION;
+import static android.Manifest.permission.ACCESS_FINE_LOCATION;
+import static android.Manifest.permission.GET_ACCOUNTS;
+import static android.Manifest.permission.WRITE_EXTERNAL_STORAGE;
+import static android.content.pm.PackageManager.PERMISSION_GRANTED;
+
+public class SplashActivity extends AppCompatActivity
+    implements BaseNewsFragment.NewsDialogListener
+{
+  public static final String[] PERMISSIONS = new String[]
+      {
+          WRITE_EXTERNAL_STORAGE,
+          ACCESS_COARSE_LOCATION,
+          ACCESS_FINE_LOCATION,
+          GET_ACCOUNTS
+      };
+  public static final String EXTRA_INTENT = "extra_intent";
+  private static final String EXTRA_ACTIVITY_TO_START = "extra_activity_to_start";
+  private static final int REQUEST_PERMISSIONS = 1;
+  private static final long DELAY = 100;
+
+  // The first launch of application ever - onboarding screen will be shown.
+  private static boolean sFirstStart;
+
+  private View mIvLogo;
+
+  private boolean mPermissionsGranted;
+  private boolean mCanceled;
+
+  @NonNull
+  private final Runnable mDelayedTask = new Runnable()
+  {
+    @Override
+    public void run()
+    {
+      init();
+//    Run delayed task because resumeDialogs() must be called after onPause()
```

Comment:
Run delayed task because resumeDialogs() must see the actual value of mCanceled flag, since onPause() callback can be blocked because of UI thread is busy with framework initialization. !?